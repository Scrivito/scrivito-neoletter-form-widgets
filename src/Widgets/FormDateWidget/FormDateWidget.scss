.scrivito-neoletter-form-widgets .form-date {
  .date-title {
    p {
      display: inline;
    }
  }
}
