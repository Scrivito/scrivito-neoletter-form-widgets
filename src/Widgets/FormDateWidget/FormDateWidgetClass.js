import * as Scrivito from "scrivito";

export const FormDateWidget = Scrivito.provideWidgetClass("FormDateWidget", {
  attributes: {
    title: "string",
    helpText: "html",
    required: "boolean",
    customFieldName: "string",
    dateType: [
      "enum",
      {
        values: ["date", "datetime-local"],
      },
    ],
  },
});
