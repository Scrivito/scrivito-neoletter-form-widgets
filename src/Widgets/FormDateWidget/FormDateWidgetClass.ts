import * as Scrivito from "scrivito";

export const FormDateWidget = Scrivito.provideWidgetClass("FormDateWidget", {
  attributes: {
    title: "html",
    helpText: "html",
    required: "boolean",
    customFieldName: "string",
    dateType: [
      "enum",
      {
        values: ["date", "datetime-local"]
      }
    ]
  }
});
