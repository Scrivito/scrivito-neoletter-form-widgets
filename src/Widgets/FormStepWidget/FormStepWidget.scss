.step-preview-count {
  position: absolute;
  top: 6px;
  left: 50%;
  background: white;
  transform: translate(-50%);
  padding: 0 5px;
  margin: 0;
}

.step-border {
  border: 2px black dotted;
}
