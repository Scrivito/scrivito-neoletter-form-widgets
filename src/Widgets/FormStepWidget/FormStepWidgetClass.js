import * as Scrivito from "scrivito";
const allowedContent = [
  "AddressWidget",
  "BlogOverviewWidget",
  "BoxWidget",
  "ButtonWidget",
  "CarouselWidget",
  "ColumnContainerWidget",
  "DividerWidget",
  "EventOverviewWidget",
  "FactWidget",
  "FeaturePanelWidget",
  "FormCheckboxWidget",
  "FormDateWidget",
  "FormHiddenFieldWidget",
  "FormConditionalContainerWidget",
  "FormInputFieldWidget",
  "FormRatingWidget",
  "FormSelectWidget",
  "GalleryWidget",
  "GoogleMapsWidget",
  "GroupWidget",
  "HeadlineWidget",
  "IconContainerWidget",
  "IconWidget",
  "ImageWidget",
  "JobOverviewWidget",
  "LinkContainerWidget",
  "PricingWidget",
  "SpaceWidget",
  "TableWidget",
  "TestimonialSliderWidget",
  "TextWidget",
  "ThumbnailGalleryWidget",
  "TickListWidget",
  "VideoWidget",
  "VimeoVideoWidget",
  "YoutubeVideoWidget",
];

export const FormStepWidget = Scrivito.provideWidgetClass("FormStepWidget", {
  onlyInside: "FormContainerWidget",
  attributes: {
    items: ["widgetlist", { only: allowedContent }],
    stepNumber: "integer",
    isActive: "boolean",
  },
});
