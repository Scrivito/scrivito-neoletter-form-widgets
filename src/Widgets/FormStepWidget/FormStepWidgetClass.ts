import * as Scrivito from "scrivito";

export const FormStepWidget = Scrivito.provideWidgetClass("FormStepWidget", {
  onlyInside: "FormStepContainerWidget",
  attributes: {
    items: "widgetlist",
    stepNumber: "integer",
    isSingleStep: "boolean"
  }
});
