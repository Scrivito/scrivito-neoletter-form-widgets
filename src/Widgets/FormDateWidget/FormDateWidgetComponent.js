import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../FormContainerWidget/utils/getFieldName";
import { HelpText } from "../FormContainerWidget/components/HelpTextComponent";
import { Mandatory } from "../FormContainerWidget/components/MandatoryComponent";
import "./FormDateWidget.scss";

Scrivito.provideComponent("FormDateWidget", ({ widget }) => {
  const [value, setValue] = React.useState("");
  const onChangeValue = (e) => {
    const d = new Date(e.target.value);
    setValue(d.toISOString());
  };
  return (
    <div className="form-date mb-3">
      <div className="date-title">
        <span className="text-super">{widget.get("title")}</span>
        {widget.get("required") && <Mandatory />}
        {widget.get("helpText") && <HelpText widget={widget} />}
      </div>

      <input
        onChange={onChangeValue}
        className="datepicker"
        type={widget.get("dateType")}
        required={widget.get("required")}
      ></input>
      <input type="hidden" name={getFieldName(widget)} value={value}></input>
    </div>
  );
});
