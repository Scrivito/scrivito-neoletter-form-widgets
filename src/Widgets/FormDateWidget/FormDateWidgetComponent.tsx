import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../FormStepContainerWidget/utils/getFieldName";
import { HelpText } from "../FormStepContainerWidget/components/HelpTextComponent";
import { Mandatory } from "../FormStepContainerWidget/components/MandatoryComponent";
import { FormDateWidget } from "./FormDateWidgetClass";
import "./FormDateWidget.scss";

Scrivito.provideComponent(FormDateWidget, ({ widget }) => {
  const [value, setValue] = React.useState("");
  const onChangeValue = (e: React.BaseSyntheticEvent) => {
    const d = new Date(e.target.value);
    setValue(d.toISOString());
  };
  return (
    <div className="form-date mb-3">
      <div className="date-title">
        <span className="text-super">{widget.get("title")}</span>
        {widget.get("required") && <Mandatory />}
        {widget.get("helpText") && <HelpText widget={widget} />}
      </div>

      <input
        onChange={onChangeValue}
        className="datepicker"
        type={widget.get("dateType")!}
        required={widget.get("required")}></input>
      <input
        type="hidden"
        className="show-in-review"
        name={getFieldName(widget)}
        value={value}></input>
    </div>
  );
});
