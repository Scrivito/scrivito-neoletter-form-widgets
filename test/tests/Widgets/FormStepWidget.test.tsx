import * as Scrivito from "scrivito";
import { FormStepWidget } from "../../../src/Widgets/FormStepWidget/FormStepWidgetClass";
import { FormDateWidget } from "../../../src/Widgets/FormDateWidget/FormDateWidgetClass";
import "../../../src/Widgets/FormStepWidget/FormStepWidgetComponent";
import "../../../src/Widgets/FormDateWidget/FormDateWidgetComponent";
import PageRenderer from "../../helpers/pageRenderer";

Scrivito.configure({ tenant: "inMemory" });

const pageRenderer = new PageRenderer();
const widgetProps = {
  items: [],
  stepNumber: 1,
  isSingleStep: true
};

describe("FormStepWidget", () => {

  it("renders without editing preview border", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(false);
    const items = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];
    pageRenderer.render({
      body: [new FormStepWidget({ ...widgetProps, items: items })]
    });

    const step = document.querySelector("div[data-step-number]");
    expect(step).not.toHaveClass("step-border");
  });

  it("renders with editing preview border", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(true);
    const items = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];
    pageRenderer.render({
      body: [new FormStepWidget({ ...widgetProps, items: items })]
    });
    const step = document.querySelector("div[data-step-number]");
    expect(step).toHaveClass("step-border");
  });
  it("renders correctly", () => {
    const items = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];

    const tree = pageRenderer.getAsJSON({
      body: [new FormStepWidget({ ...widgetProps, items: items })]
    });
    expect(tree).toMatchSnapshot();
  });
});
