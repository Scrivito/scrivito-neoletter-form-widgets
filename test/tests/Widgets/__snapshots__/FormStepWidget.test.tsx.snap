// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStepWidget renders correctly 1`] = `
<div>
  <div>
    <div
      className="step-border "
      data-step-number={0}
    >
      <span
        className="step-preview-count"
      >
        Step 0
      </span>
      <div
        className="row"
      >
        <div>
          <div>
            <div
              className="form-date mb-3"
            >
              <div
                className="date-title"
              >
                <span
                  className="text-super"
                >
                  date
                </span>
              </div>
              <input
                className="datepicker"
                onChange={[Function]}
                required={false}
                type="date"
              />
              <input
                className="show-in-review"
                name="custom_date"
                type="hidden"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
